plugin/keyring_vault/CMakeLists.txt

```diff
@@ -49,8 +48,10 @@ MYSQL_ADD_PLUGIN(keyring_vault
                  MODULE_ONLY
                  MODULE_OUTPUT_NAME "keyring_vault")
 
-get_property(keyring_vault_link_flags TARGET keyring_vault PROPERTY LINK_FLAGS)
-set_property(TARGET keyring_vault PROPERTY LINK_FLAGS "${keyring_vault_link_flags} -Wl,--version-script=${CMAKE_SOURCE_DIR}/plugin/keyring_vault/keyring_vault.version")
+IF(LINK_FLAG_NO_UNDEFINED)
```

Comment:
Maybe a word of comment that LINK_FLAG_NO_UNDEFINED is defined for Linux system only ? What if confusing about this flag that only libmysql and libmysqld actually use it, all other plugins export undefined symbols. This is optional comment and might be skipped.

Reply:
As discussed on Slack, pushing with "# We limit symbols exported on Linux to only those required by server." added